dataflow_test.py

```diff
@@ -523,3 +523,71 @@ def test_count_filter_counts_correctly():
 
     assert result.n_passed == len( odd_numbers)
     assert result.n_failed == len(even_numbers)
+
+
+@parametrize("component",
+             (df.map   (lambda x: x)    ,
+              df.filter(lambda x: x > 0),
+              df.sink  (print)          ,
+              df.branch(df.sink(print)) ,
+              df.pipe  (df.map(abs))    ))
+def test_string_to_pick_ignores_components(component):
+    assert component is df.string_to_pick(component)
+
+
+def test_string_to_pick():
+
+    # string_to_pick creates a pipe component that picks
+    # an item from the namespace and pushes it through the pipe
```

Comment:
This comment assumes that the user knows that there is (well ... might be) a namespace flowing through the pipe. I don't think this has been mentioned anywhere yet.

In general, I think that we'll need to think carefully about how we introduce the idea of the namespace flowing though the pipes, and all the features that exist purely for working with these namespaces.

Reply:
Well, `string_to_pick` is specifically introduced to work with that feature, so it must assume it exists, right? But the documentation needs to be there beforehand, I agree.

Reply:
Yup. I think there are a few things to be done here:

1. Don't mention `string_to_pick` at all in the user-facing docs (which these tests are, at least for the time being), only show the use of strings at the front of pipes.

2. Ensure that the namespace flow is discussed before this.

3. Rename `string_to_pick` to `_string_to_pick` (underscore).

4. Test `_string_to_pick` in a non-user-facing unit tests, if deemed necessary.

5. Think about whether we want to admit tuples-to-pick in addition to a single string.

Reply:
OK. Do you want to address all that in this PR? I would certainly leave the last point for a different PR.

Reply:
Given that I'm still going to be rebasing `master` like crazy, for a while, I'm reluctant to accept any PR for the time being. So we have some time to think about this.

The last point is definitely not for this PR, agreed. The others, might make sense in here.